x-pack/qa/full-cluster-restart/src/test/java/org/elasticsearch/xpack/restart/FullClusterRestartIT.java

```diff
@@ -254,6 +256,77 @@ public void testWatcher() throws Exception {
         }
     }
 
+    /**
+     * Tests that a RollUp job created on a old cluster is correctly restarted after the upgrade.
+     */
+    public void testRollupAfterRestart() throws Exception {
+        assumeTrue("Rollup can be tested with 6.3.0 and onwards", oldClusterVersion.onOrAfter(Version.V_6_3_0));
+        if (runningAgainstOldCluster) {
+            final int numDocs = 59;
+            final int year = randomIntBetween(1970, 2018);
+
+            // index documents for the rollup job
+            final StringBuilder bulk = new StringBuilder();
+            for (int i = 0; i < numDocs; i++) {
+                bulk.append("{\"index\":{\"_index\":\"rollup-docs\",\"_type\":\"doc\"}}\n");
+                String date = String.format(Locale.ROOT, "%04d-01-01T00:%02d:00Z", year, i);
+                bulk.append("{\"timestamp\":\"").append(date).append("\",\"value\":").append(i).append("}\n");
+            }
+            bulk.append("\r\n");
+
+            final Request bulkRequest = new Request("POST", "/_bulk");
+            bulkRequest.setJsonEntity(bulk.toString());
+            client().performRequest(bulkRequest);
+
+            // create the rollup job
+            final Request createRollupJobRequest = new Request("PUT", "/_xpack/rollup/job/rollup-job-test");
+            createRollupJobRequest.setJsonEntity("{"
+                    + "\"index_pattern\":\"rollup-*\","
+                    + "\"rollup_index\":\"results-rollup\","
+                    + "\"cron\":\"*/30 * * * * ?\","
+                    + "\"page_size\":100,"
+                    + "\"groups\":{"
+                    + "    \"date_histogram\":{"
+                    + "        \"field\":\"timestamp\","
+                    + "        \"interval\":\"5m\""
+                    + "      }"
+                    + "},"
+                    + "\"metrics\":["
+                    + "    {\"field\":\"value\",\"metrics\":[\"min\",\"max\",\"sum\"]}"
+                    + "]"
+                    + "}");
+
+            Map<String, Object> createRollupJobResponse = toMap(client().performRequest(createRollupJobRequest));
+            assertThat(createRollupJobResponse.get("acknowledged"), equalTo(Boolean.TRUE));
+
+            // start the rollup job
+            final Request startRollupJobRequest = new Request("POST", "_xpack/rollup/job/rollup-job-test/_start");
```

Comment:
Technically, you don't have to start the job if you just want to make sure the task is running.  We start the persistent task when the job is created in the CreateRollupJob API.

Although I wonder if there might be a potentially rare timing issue here?  The CreateRollupJob API returns when the persistent task framework acknowledges that task was created.  But I believe there might be a lag between that and when the allocated task is actually created on the target node?  

So it might be possible for the StartJob API (or `assertRollupJob()` if starting is skipped) to fail because the allocated task hasn't started yet?  Maybe we need an `assertBusy` checking for the job, like below?

I may be wrong about that though, you're much more familiar with how the persistent tasks work :)

Reply:
> Technically, you don't have to start the job if you just want to make sure the task is running. We start the persistent task when the job is created in the CreateRollupJob API.

Ok, I've been fooled by [the doc](https://www.elastic.co/guide/en/elasticsearch/reference/6.3/rollup-put-job.html).



> Although I wonder if there might be a potentially rare timing issue here?

Good point. I'll take a look at this, thanks.

Reply:
Ah yeah, the docs are a little confusing with regards to how it works internally.  The persistent/allocated Task is always running, start and stop just toggles internal state.  If the task doesn't exist, the job doesn't exist basically.

Reply:
I agree. The rollup job is started so that we can ensure that it is correctly restarted after upgrade: it means that the state was correctly persisted on the old cluster and correctly deserialized back on the upgraded cluster.